tools/CMakeLists.txt

```diff
@@ -219,119 +113,283 @@ else()
+  target_link_libraries(pcl_icp2d pcl_common pcl_io pcl_registration)
+
+  PCL_ADD_EXECUTABLE(pcl_elch COMPONENT ${SUBSYS_NAME} SOURCES elch.cpp)
+  target_link_libraries(pcl_elch pcl_common pcl_io pcl_registration)
+
+  PCL_ADD_EXECUTABLE(pcl_lum COMPONENT ${SUBSYS_NAME} SOURCES lum.cpp)
+  target_link_libraries(pcl_lum pcl_common pcl_io pcl_registration)
+
+  PCL_ADD_EXECUTABLE(pcl_ndt2d COMPONENT ${SUBSYS_NAME} SOURCES ndt2d.cpp)
+  target_link_libraries(pcl_ndt2d pcl_common pcl_io pcl_registration)
+
+  PCL_ADD_EXECUTABLE(pcl_ndt3d COMPONENT ${SUBSYS_NAME} SOURCES ndt3d.cpp)
+  target_link_libraries(pcl_ndt3d pcl_common pcl_io pcl_registration)
+
+  PCL_ADD_EXECUTABLE(pcl_transform_point_cloud COMPONENT ${SUBSYS_NAME} SOURCES transform_point_cloud.cpp)
+  target_link_libraries (pcl_transform_point_cloud pcl_common pcl_io pcl_registration)
+
+  PCL_ADD_EXECUTABLE(pcl_transform_from_viewpoint COMPONENT ${SUBSYS_NAME} SOURCES transform_from_viewpoint.cpp)
+  target_link_libraries (pcl_transform_from_viewpoint pcl_common pcl_io pcl_registration)
+endif()
+
+if(TARGET pcl_recognition)
+
+  PCL_ADD_EXECUTABLE(pcl_train_linemod_template COMPONENT ${SUBSYS_NAME} SOURCES train_linemod_template.cpp)
+  target_link_libraries(pcl_train_linemod_template pcl_common pcl_io pcl_segmentation pcl_recognition)
+
+  PCL_ADD_EXECUTABLE(pcl_match_linemod_template COMPONENT ${SUBSYS_NAME} SOURCES match_linemod_template.cpp)
+  target_link_libraries(pcl_match_linemod_template pcl_common pcl_io pcl_recognition)
+
+  PCL_ADD_EXECUTABLE(pcl_linemod_detection COMPONENT ${SUBSYS_NAME} SOURCES linemod_detection.cpp)
+  target_link_libraries(pcl_linemod_detection pcl_common pcl_io pcl_recognition)
+
+endif()
+
+
+if(TARGET pcl_visualization)
+  PCL_ADD_EXECUTABLE(pcl_registration_visualizer COMPONENT ${SUBSYS_NAME} SOURCES registration_visualizer.cpp)
+  target_link_libraries(pcl_registration_visualizer pcl_common pcl_io pcl_kdtree pcl_filters pcl_registration pcl_visualization)
+
+  PCL_ADD_EXECUTABLE(pcl_octree_viewer COMPONENT ${SUBSYS_NAME} SOURCES octree_viewer.cpp)
+  target_link_libraries(pcl_octree_viewer pcl_common pcl_io pcl_octree pcl_visualization pcl_kdtree pcl_filters)
+
+  PCL_ADD_EXECUTABLE(pcl_mesh2pcd COMPONENT ${SUBSYS_NAME} SOURCES mesh2pcd.cpp)
+  target_link_libraries(pcl_mesh2pcd pcl_common pcl_io pcl_visualization pcl_filters)
+
+  PCL_ADD_EXECUTABLE(pcl_mesh_sampling COMPONENT ${SUBSYS_NAME} SOURCES mesh_sampling.cpp)
+  target_link_libraries(pcl_mesh_sampling pcl_common pcl_io pcl_visualization pcl_filters)
+
+  PCL_ADD_EXECUTABLE(pcl_virtual_scanner COMPONENT ${SUBSYS_NAME} SOURCES virtual_scanner.cpp)
+  target_link_libraries(pcl_virtual_scanner pcl_common pcl_io pcl_filters pcl_visualization)
+
+  PCL_ADD_EXECUTABLE(pcl_voxel_grid_occlusion_estimation COMPONENT ${SUBSYS_NAME} SOURCES voxel_grid_occlusion_estimation.cpp)
+  target_link_libraries (pcl_voxel_grid_occlusion_estimation pcl_common pcl_io pcl_filters pcl_visualization)
```

Comment:
I think we should also check for `pcl_registration` (tool `pcl_registration_visualizer`) and `pcl_filters` (all marked tools)